### PR TITLE
Changed arg parsing. Checked for valid inputs. Minor refactoring

### PR DESCRIPTION
**Pull request by LTAcosta**

### CODE REVIEW

File: spims [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
 
-import sys, getopt, datetime
+import sys, getopt, datetime, imghdr, optparse
 from PIL import Image
 from PIL import ImageChops
 
-def matchTemplate(searchImage, templateImage, searchName, templateName):
+def matchTemplate(patternArg, sourceArg):
+    try:
+        validTypes = ['gif', 'jpeg', 'png']
```
Comment (lernerb) [inline spims]: Should make this whole try a helper that gets called

**Comment by LTAcosta** [comment]
Right. I'll change that now.

**Comment by LTAcosta** [comment]
There, now all the argument checks are in a separate function. It does make things cleaner.

File: spims [hunk 2]
```diff
@@ -1,24 +1,32 @@
 #!/usr/bin/env python
 
-import sys, getopt, datetime
+import sys, getopt, datetime, imghdr, optparse
```
Comment (lernerb) [inline spims]: Just for me to remember, we don't use datetime anymore.

File: spims [hunk 3]
```diff
@@ -27,32 +35,47 @@ def matchTemplate(searchImage, templateImage, searchName, templateName):
...
-        print "%s matches %s at %ix%i+%i+%i" % (templateName, searchName, templateWidth, templateHeight, matching_xs, matching_ys)
- 
+    #TODO: Figure out what number to put here
+    if minScore > 0:    
+        print "%s matches %s at %ix%i+%i+%i" % (patternName, sourceName, patternWidth, patternHeight, matching_xs, matching_ys)
+    
+    # Function completed sucessfully    
+    return 0
+
+def isInputValid(patternArg, sourceArg):
```
Comment (lernerb) [inline spims:58]: We should figure out the proper 'javadocs' style for Python and add this to both functions

File: spims [hunk 4]
```diff
@@ -1,24 +1,32 @@
 #!/usr/bin/env python
 
-import sys, getopt, datetime
+import sys, getopt, datetime, imghdr, optparse
 from PIL import Image
 from PIL import ImageChops
 
-def matchTemplate(searchImage, templateImage, searchName, templateName):
```
Comment (lernerb) [inline spims:7]: We should figure out the proper 'javadocs' style for Python and add this to both functions

File: spims (hunk 2 above)
Comment (lernerb) [inline spims]: Imports should usually be on separate lines, e.g.:

Yes: import os
     import sys

No:  import sys, os

File: spims [hunk 5]
```diff
@@ -1,24 +1,32 @@
 #!/usr/bin/env python
 
-import sys, getopt, datetime
+import sys, getopt, datetime, imghdr, optparse
 from PIL import Image
 from PIL import ImageChops
```
Comment (lernerb) [inline spims]: It's okay to say this though:

from subprocess import Popen, PIPE

File: spims (hunk 2 above)
Comment (LTAcosta) [inline spims]: That's an easy change. I can do that quickly right now

File: spims [hunk 6]
```diff
@@ -1,24 +1,32 @@
 #!/usr/bin/env python
```
Comment (lernerb) [inline spims:1]: And to add more stuff, we should add this (i'll get to it tonight)

http://www.cs.swarthmore.edu/~richardw/cs21-s09/python_codestyle.php

File: spims (hunk 3 above)
Comment (LTAcosta) [inline spims:58]: I just checked. Apparently it is three quotes ("""). I'll add these

**Comment by lernerb** [comment]
We got a bunch of stuff we can take care of tonight, I'll do whatever you odn't do lennon after I go for dinner.

File: spims (hunk 3 above)
Comment (LTAcosta) [inline spims:58]: http://www.python.org/dev/peps/pep-0257/

Comment (lernerb) [inline spims:58]: It looks like comments go inside the functions:

ex.

def square_the_biggest(n1, n2):*
"""
  Function: square_the_biggest

---

```
Returns the square of the largest of its two input values

n1: one real value
n2: the other real value

returns: the square of the larger of n1 and n2
```

""" *

On Thu, Feb 14, 2013 at 6:57 PM, Lennon Acosta notifications@github.comwrote:

> In spims:
> 
> > @@ -27,32 +35,47 @@ def matchTemplate(searchImage, templateImage, searchName, templateName):
> >                  matching_xs = xs
> >                  matching_ys = ys
> > -    if minScore > 1000:
> > 
> > ## \-        print "%s matches %s at %ix%i+%i+%i" % (templateName, searchName, templateWidth, templateHeight, matching_xs, matching_ys)
> > -    #TODO: Figure out what number to put here
> > -    if minScore > 0:
> > -        print "%s matches %s at %ix%i+%i+%i" % (patternName, sourceName, patternWidth, patternHeight, matching_xs, matching_ys)
> >   +
> > -    # Function completed sucessfully
> > -    return 0
> >   +
> >   +def isInputValid(patternArg, sourceArg):
> 
> I just checked. Apparently it is three quotes ("""). I'll add these
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lernerb/SoftwareDev/pull/8/files#r3022667.

## 

Brandon Lerner

Comment (LTAcosta) [inline spims:58]: Yeah, I saw that. While looking further, I have seen a few different ways of setting it up, so I've chosen one that looks closes to the style guide.

File: spims (hunk 6 above)
Comment (LTAcosta) [inline spims:1]: And here is the official style guide: http://www.python.org/dev/peps/pep-0008/

**Comment by LTAcosta** [comment]
Alright. I think I'm done for now. Feel free to look over those style guides and see if there's anything else that should be changes.

We should also make sure we merge this back in before we do any other big changes to the search function, since all the name changes will make it a pain.

**Comment by lernerb** [comment]
Yeah we can go ahead and merge it if we put the issues on the side,
otherwise no one do work in this file until i merge it tomorrow :P

On Thu, Feb 14, 2013 at 7:34 PM, Lennon Acosta notifications@github.comwrote:

> Alright. I think I'm done for now. Feel free to look over those style
> guides and see if there's anything else that should be changes.
> 
> We should also make sure we merge this back in before we do any other big
> changes to the search function, since all the name changes will make it a
> pain.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/lernerb/SoftwareDev/pull/8#issuecomment-13587504.

## 

Brandon Lerner

**Pull request merged**